common/views/pages/_app.tsx

```diff
@@ -83,11 +83,21 @@ const WecoApp: FunctionComponent<WecoAppProps> = ({
 
   const serverData = isServerDataSet ? pageProps.serverData : defaultServerData;
 
+  const reloadConsentStatus = () => window.location.reload();
+
   useMaintainPageHeight();
   useEffect(() => {
     document.documentElement.classList.add('enhanced');
   }, []);
 
+  useEffect(() => {
+    window.addEventListener('analyticsConsentChange', reloadConsentStatus);
+    return () => {
+      window.removeEventListener('analyticsConsentChange', reloadConsentStatus);
+      window.CookieControl.open();
```

Comment:
Do we need`window.CookieControl.open();` here?

Reply:
Yes I just removed it in the `useEffect` as it would've never reached it there but reaches it in this one.
If you mean at all, I thought it'd be a less weird behaviour to see your form again but it's definitely debatable. To be honest I just wish it was a Save button instead of acting on toggle.

Reply:
I guess I could write it down as a slight UX irritant for me, but I looked through the docs and not sure we can change that...

Reply:
Just tried this locally and I feel like the jump caused by the reload on the toggle and then the wait for the banner to slide back out is a bit jarring. It makes me think perhaps we should properly deal with consent change without reloading. Maybe best left to show @j-jaworski, not blocking for merge.

Reply:
I think most definitely making an accept / reject choice should dismiss the banner and probably not the toggle, but that's complicated by the toggle change firing the same events 😬 .

Reply:
Agreed with the jarring. Off the top of my head though I'm really unsure how to remove the scripts without reloading with the way we've implemented it, but we could investigate that next week and depend less on the cookie value in `_document`, maybe 🤔

Reply:
Ahh yes of course, it's because we're still allowing by default whereas in the future we'll reject by default. If we do move forward with this we can change the logic to match!